src/main/java/com/nhoryzon/mc/eidolon/block/TableBlockBase.java

```diff
@@ -0,0 +1,104 @@
+package com.nhoryzon.mc.eidolon.block;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Waterloggable;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.ItemPlacementContext;
+import net.minecraft.state.StateManager;
+import net.minecraft.state.property.BooleanProperty;
+import net.minecraft.state.property.Properties;
+import net.minecraft.util.function.BooleanBiFunction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.util.shape.VoxelShapes;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.WorldAccess;
+import net.minecraft.world.WorldView;
+import net.minecraft.world.tick.OrderedTick;
+import org.jetbrains.annotations.Nullable;
+
+public class TableBlockBase extends BlockBase implements Waterloggable {
+
+    private final VoxelShape base;
+    private final VoxelShape corner;
+
+    private static final BooleanProperty WATERLOGGED = Properties.WATERLOGGED;
+    private static final BooleanProperty
+            NX = BooleanProperty.of("nx"),
+            PX = BooleanProperty.of("px"),
+            NZ = BooleanProperty.of("nz"),
+            PZ = BooleanProperty.of("pz");
+
+    public TableBlockBase(Settings settings) {
+        this(settings, VoxelShapes.cuboid(0, .75, 0, 1, 1, 1));
+    }
+
+    public TableBlockBase(Settings settings, VoxelShape baseShape) {
+        super(settings);
+        base = baseShape;
+        corner = VoxelShapes.combine(base,
+                VoxelShapes.cuboid(.0625, 0, .0625, .9375, .75, .9375), BooleanBiFunction.OR);
+        setDefaultState(getStateManager().getDefaultState()
+                .with(NX, false).with(PX, false).with(NZ, false).with(PZ, false));
+    }
+
+    @Override
+    public VoxelShape getRaycastShape(BlockState state, BlockView world, BlockPos pos) {
+        boolean nx = state.get(NX), px = state.get(PX), nz = state.get(NZ), pz = state.get(PZ);
+
+        return (!nx && !nz) || (!nx && !pz) || (!px && !pz) || (!px && !nz) ? corner : base;
+    }
+
+    @Nullable
+    @Override
+    public BlockState getPlacementState(ItemPlacementContext ctx) {
+        WorldView world = ctx.getWorld();
+        BlockPos blockPos = ctx.getBlockPos();
+
+        return updateCorners(world, blockPos, super.getPlacementState(ctx))
+                .with(WATERLOGGED, ctx.getWorld().getFluidState(ctx.getBlockPos()).getFluid() == Fluids.WATER);
```

Comment:
`ctw.getWorld()` et `ctx.getBlockPos()` sont stockés dans des variables mais ces variables ne sont pas utilisées dans le `.with` et du coup tu refais appel aux méthodes du context

Reply:
Fix in c87eccbc50e6730dfe930b7d0ca78777bff8ff3f.